flojoy/flojoy_python.py

```diff
@@ -229,19 +238,51 @@ def wrapper(**kwargs: flojoyKwargs):
             func_params["node_id"] = node_id
             func_params["job_id"] = job_id
 
-            print("executing node_id:", node_id, "previous_job_ids:", previous_job_ids)
+            print("executing node_id:", node_id, "previous_jobs:", previous_jobs)
             print(node_id, " params: ", json.dumps(func_params, indent=2))
-            node_inputs = fetch_inputs(previous_job_ids, mock)
+            node_inputs, dict_inputs = fetch_inputs(previous_jobs)
+
+            # constructing the inputs
+            print(f"constructing inputs for {func}")
+            args = {}
+            sig = signature(func)
+
+            # once all the nodes are migrated to the new node api, remove the if condition
+            keys = list(sig.parameters)
+            if (
+                len(sig.parameters) == 2
+                and sig.parameters[keys[0]].annotation == list[DataContainer]
+            ):
+                args[keys[0]] = node_inputs
+            else:
+                args = {**args, **dict_inputs}
```

Comment:
isn't args always empty here? why do we need to destructure it?

Reply:
To keep it uniform and less error prone really so in future we don’t forget to add the destructuring  here should some other code adds to args before it